### ISSUE TITLE
linter: --rules omits vitest equivalents of shared jest/vitest rules

### ISSUE THREAD

**Issue by christopher-buss** [oxc-project/oxc#20466]
## Summary

When both `jest` and `vitest` plugins are enabled, `oxlint --rules` only lists shared rules under `jest`, omitting the `vitest/*` variant entirely. This makes it impossible to programmatically discover the full set of native vitest rule names from CLI output.

## Reproduction

```bash
oxlint --rules
```

Rules like `no-conditional-expect`, `no-alias-methods`, `valid-expect`, `no-identical-title`, etc. only appear as `jest/*`:

```
| no-conditional-expect | jest   |
| no-alias-methods      | jest   |
| valid-expect          | jest   |
```

The corresponding `vitest/no-conditional-expect`, `vitest/no-alias-methods`, `vitest/valid-expect` entries are missing from the output.

Vitest-only rules (no jest equivalent) do appear correctly:

```
| consistent-each-for   | vitest |
| hoisted-apis-on-top   | vitest |
```

## Impact

We use `--rules` output to generate a TypeScript union type of all native oxlint rule names (`OxlintNativeRuleName`). Because shared rules only appear under `jest`, our generated type is missing ~40 `vitest/*` entries, which means we can't type-check vitest rule configs against native oxlint support.

## Expected behavior

Shared rules should appear once per plugin that implements them:

```
| no-conditional-expect | jest   |
| no-conditional-expect | vitest |
```

Or alternatively, a structured output format (JSON) that maps rules to all implementing plugins.

## Related

- #19486 — proposes exporting structured rule metadata as JS objects (would also solve this)

**Comment by camchenry** [comment]
This should be fixed in the next release. In #21664 we split all of the Vitest rules into separate rules from the Jest ones so that we can develop them independently. 1.62 had a partial version of this so you can start to see some of the rules there, but 1.63 should have the full list of vitest rules. Here's a partial list from 1.62 (for example):

```
| consistent-each-for                                 | vitest     |         |          |          |
| expect-expect                                       | vitest     |         |          |          |
| hoisted-apis-on-top                                 | vitest     |         |          | 💡       |
| no-conditional-expect                               | vitest     |         |          |          |
| no-conditional-tests                                | vitest     |         |          |          |
| no-disabled-tests                                   | vitest     |         |          |          |
| no-focused-tests                                    | vitest     |         |          | 🛠️        |
| require-awaited-expect-poll                         | vitest     |         |          |          |
| require-local-test-context-for-concurrent-snapshots | vitest     |         |          |          |
| require-mock-type-parameters                        | vitest     |         |          |          |
| valid-expect                                        | vitest     |         |          | 💡       |
| valid-title                                         | vitest     |         |          | 🛠️        |
| warn-todo                                           | vitest     |         |          |          |
```
